client/src/components/SignUpForm/SignUpForm.tsx

```diff
@@ -0,0 +1,150 @@
+import { Field, Form, Formik } from 'formik';
+import * as Yup from 'yup';
+import React, { useState } from 'react';
+import styles from './SignUpForm.module.scss';
+import { SignUpDto } from '../../dtos/SignUp.dto';
+import { CustomButton } from '../common/CustomButton/CustomButton';
+import { CustomInput } from '../common/CustomInput/CustomInput';
+import { useAppDispatch } from '../../hooks/redux';
+import { signUp } from '../../redux/actionCreators/auth';
+import { Preloader } from '../common/Preloader/Preloader';
+
+interface IProps {}
+
+const phoneRegExp = /^[0-9]{3}[0-9]{2}[0-9]{3}[0-9]{4,5}$/;
+
+export const SignUpForm: React.FC<IProps> = ({}) => {
+    const dispatch = useAppDispatch();
+    const [formSendError, setFormSendError] = useState('');
+
+    const formInitialValues: SignUpDto = {
+        email: '',
+        password: '',
+        firstName: '',
+        lastName: '',
+        phone: '',
+    };
+
+    return (
+        <Formik
+            enableReinitialize={true}
+            initialValues={formInitialValues}
+            onSubmit={async (values, { resetForm }) => {
```

Comment:
It is also great thing to do is to move callbacks to separate functions from render.